jmix-flowui/flowui/src/main/java/io/jmix/flowui/xml/layout/support/PrefixSuffixLoaderSupport.java

```diff
@@ -52,58 +57,34 @@ public void setApplicationContext(ApplicationContext applicationContext) throws
 
     public void createPrefixSuffixComponents(Component component, Element element) {
         if (component instanceof HasPrefix hasPrefixComponent) {
-            createPrefixComponent(hasPrefixComponent, element);
+            initPrefixSuffixComponent(element.element("prefix"), hasPrefixComponent::setPrefixComponent);
         }
 
         if (component instanceof HasSuffix hasSuffixComponent) {
-            createSuffixComponent(hasSuffixComponent, element);
+            initPrefixSuffixComponent(element.element("suffix"), hasSuffixComponent::setSuffixComponent);
         }
     }
 
     public void loadPrefixSuffixComponents() {
-        if (pendingLoadPrefixComponent != null) {
-            pendingLoadPrefixComponent.loadComponent();
-        }
-
-        if (pendingLoadSuffixComponent != null) {
-            pendingLoadSuffixComponent.loadComponent();
-        }
+        pendingLoadComponents.forEach(ComponentLoader::loadComponent);
+        pendingLoadComponents.clear();
     }
 
-    protected void createPrefixComponent(HasPrefix hasPrefixComponent, Element hasPrefixElement) {
-        Element prefix = hasPrefixElement.element("prefix");
-
-        if (prefix == null) {
-            return;
-        }
-
-        if (prefix.elements().size() != 1) {
-            throw new GuiDevelopmentException("Only one prefix component can be defined", context);
-        }
-
-        Element prefixComponentElement = prefix.elements().get(0);
-        pendingLoadPrefixComponent = getLayoutLoader().createComponentLoader(prefixComponentElement);
-        pendingLoadPrefixComponent.initComponent();
-
-        hasPrefixComponent.setPrefixComponent(pendingLoadPrefixComponent.getResultComponent());
-    }
-
-    protected void createSuffixComponent(HasSuffix hasSuffix, Element hasSuffixElement) {
-        Element suffix = hasSuffixElement.element("suffix");
-
-        if (suffix == null) {
+    protected void initPrefixSuffixComponent(@Nullable Element element, Consumer<Component> setter) {
+        if (element == null) {
             return;
         }
 
-        if (suffix.elements().size() != 1) {
+        if (element.elements().size() != 1) {
             throw new GuiDevelopmentException("Only one suffix component can be defined", context);
```

Comment:
It seems that it's better to pass element name to the method so the exception text adapts to element name. Currently it's hardcoded `suffix`